### ISSUE TITLE
Split user data from maintained data

### ISSUE THREAD

**Issue by matt-chan** [Azure/az-hop#1444]
## In what area(s)?

> /area administration
> /area ansible
> /area autoscaling
> /area configuration
> /area cyclecloud
> /area documentation
> /area image
> /area job-scheduling
> /area monitoring
> /area ood
> /area remote-visualization
> /area user-management

## Describe the feature

Can we split the storage of user data from admin maintained data? This would allow us to perform upgrades without service disruptions for users. It also means we can reason better about which files can be removed without data loss and which cannot. 

For example, in the current config, slurm is stored in {{homedir_mountpoint}}/slurm. This means that if we wanted to migrate user data during a slurm version upgrade, we would have to mount ANF in another cluster, and then overwrite the slurm data. This would break slurm in the old deployment.

Instead, if we stored slurm in a second partition (e.g. /software), we could mount /anfhome on the new cluster and the old cluster simultaneously, and make changes to the new cluster's copy of /software without disrupting users on the old cluster. When the upgrades are done, we could move users to the new cluster and they could use it immediately.

**Comment by matt-chan** [comment]
Does every worker need to have access to the slurm file system? If so, I wonder if using azure files would be a better choice. The smallest premium volume (100G) should be about $20/month. 

We also have a bunch of other ephemeral files (spack binaries, etc) so we could place everything on the same file system if that were the case.

**Comment by xpillons** [comment]
yes every worker need to access the slurm configuration files. We may have to do scale testing as Azure Files has a higher latency than ANF or a DNS server, so we want to make sure we can sustain the load.
